crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -121,11 +121,11 @@ impl SargonOS {
 
         // Extracting the entities requiring auth to check if the notary is signatory
         let profile = self.profile_state_holder.profile()?;
+        let summary = manifest
+            .summary()
```

Comment:
so `fn summary` returns an `Option`? Dont we want it to return `Result` instead?